Tests/UnitTests/Plugins/Geant4/Geant4SurfaceProviderTests.cpp

```diff
@@ -0,0 +1,302 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <boost/test/unit_test.hpp>
+
+#include "Acts/Detector/LayerStructureBuilder.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Plugins/Geant4/Geant4SurfaceProvider.hpp"
+#include "Acts/Utilities/BinningType.hpp"
+
+#include <filesystem>
+#include <memory>
+#include <string>
+
+#include "G4Box.hh"
+#include "G4GDMLParser.hh"
+#include "G4LogicalVolume.hh"
+#include "G4NistManager.hh"
+#include "G4PVPlacement.hh"
+#include "G4RotationMatrix.hh"
+#include "G4SystemOfUnits.hh"
+#include "G4ThreeVector.hh"
+#include "G4Transform3D.hh"
+
+class G4VPhysicalVolume;
+
+int nLayers = 5;
+int nChips = 5;
+int nArms = 2;
+
+double cellDimX = 0.4 * cm;
+double cellDimY = 0.3 * cm;
+double cellDimZ = 0.1 * cm;
+
+double armOffset = 10 * cm;
```

Comment:
Maybe we could set them all `const` to clarify, that we use the same parameters for all tests.
```suggestion
const int nLayers = 5;
const int nChips = 5;
const int nArms = 2;

const double cellDimX = 0.4 * cm;
const double cellDimY = 0.3 * cm;
const double cellDimZ = 0.1 * cm;

const double armOffset = 10 * cm;
```